src/datadog/datadog_agent.cpp

```diff
@@ -211,10 +265,21 @@ void DatadogAgent::flush() {
 
   // This is the callback for the HTTP response.  It's invoked
   // asynchronously.
-  auto on_response = [samplers = std::move(response_handlers),
+  auto on_response = [this, samplers = std::move(response_handlers),
```

Comment:
I'm thinking to err on the side of `[telemetry = tracer_telemetry_, ...]` instead of `[this, ...]`.

On the other hand, we can go all in on "the DatadogAgent will outlive the HTTPClient" and use `this` everywhere, without copying any of `*this`'s members. The lifetimes are correct.

The reason I prefer the unnecessary `shared_ptr` copies is that it might aid in a future refactoring of `class Curl` into different "event loop" backends.

Anyway, let's go with one or the other. Which is up to you.